internal/services/appservice/helpers/web_app_slot_schema.go

```diff
@@ -518,3 +781,262 @@ func FlattenSiteConfigLinuxWebAppSlot(appSiteSlotConfig *web.SiteConfig, healthC
+		ManagedPipelineMode:      string(appSiteSlotConfig.ManagedPipelineMode),
+		MinTlsVersion:            string(appSiteSlotConfig.MinTLSVersion),
+		WorkerCount:              int(utils.NormaliseNilableInt32(appSiteSlotConfig.NumberOfWorkers)),
+		RemoteDebugging:          utils.NormaliseNilableBool(appSiteSlotConfig.RemoteDebuggingEnabled),
+		RemoteDebuggingVersion:   strings.ToUpper(utils.NormalizeNilableString(appSiteSlotConfig.RemoteDebuggingVersion)),
+		ScmIpRestriction:         FlattenIpRestrictions(appSiteSlotConfig.ScmIPSecurityRestrictions),
+		ScmMinTlsVersion:         string(appSiteSlotConfig.ScmMinTLSVersion),
+		ScmType:                  string(appSiteSlotConfig.ScmType),
+		ScmUseMainIpRestriction:  utils.NormaliseNilableBool(appSiteSlotConfig.ScmIPSecurityRestrictionsUseMain),
+		Use32BitWorker:           utils.NormaliseNilableBool(appSiteSlotConfig.Use32BitWorkerProcess),
+		UseManagedIdentityACR:    utils.NormaliseNilableBool(appSiteSlotConfig.AcrUseManagedIdentityCreds),
+		VirtualApplications:      flattenVirtualApplications(appSiteSlotConfig.VirtualApplications),
+		WebSockets:               utils.NormaliseNilableBool(appSiteSlotConfig.WebSocketsEnabled),
+		VnetRouteAllEnabled:      utils.NormaliseNilableBool(appSiteSlotConfig.VnetRouteAllEnabled),
+	}
+
+	if appSiteSlotConfig.APIManagementConfig != nil && appSiteSlotConfig.APIManagementConfig.ID != nil {
+		siteConfig.ApiManagementConfigId = *appSiteSlotConfig.APIManagementConfig.ID
+	}
+
+	if appSiteSlotConfig.APIDefinition != nil && appSiteSlotConfig.APIDefinition.URL != nil {
+		siteConfig.ApiDefinition = *appSiteSlotConfig.APIDefinition.URL
+	}
+
+	if appSiteSlotConfig.DefaultDocuments != nil {
+		siteConfig.DefaultDocuments = *appSiteSlotConfig.DefaultDocuments
+	}
+
+	if appSiteSlotConfig.NumberOfWorkers != nil {
+		siteConfig.WorkerCount = int(*appSiteSlotConfig.NumberOfWorkers)
+	}
+
+	var winAppStack ApplicationStackWindows
+	winAppStack.NetFrameworkVersion = utils.NormalizeNilableString(appSiteSlotConfig.NetFrameworkVersion)
+	winAppStack.PhpVersion = utils.NormalizeNilableString(appSiteSlotConfig.PhpVersion)
+	winAppStack.NodeVersion = utils.NormalizeNilableString(appSiteSlotConfig.NodeVersion)
+	winAppStack.PythonVersion = utils.NormalizeNilableString(appSiteSlotConfig.PythonVersion)
+	winAppStack.JavaVersion = utils.NormalizeNilableString(appSiteSlotConfig.JavaVersion)
+	winAppStack.JavaContainer = utils.NormalizeNilableString(appSiteSlotConfig.JavaContainer)
+	winAppStack.JavaContainerVersion = utils.NormalizeNilableString(appSiteSlotConfig.JavaContainerVersion)
+
+	siteConfig.WindowsFxVersion = utils.NormalizeNilableString(appSiteSlotConfig.WindowsFxVersion)
+	if siteConfig.WindowsFxVersion != "" {
+		// Decode the string to docker values
+		parts := strings.Split(strings.TrimPrefix(siteConfig.WindowsFxVersion, "DOCKER|"), ":")
+		winAppStack.DockerContainerTag = parts[1]
```

Comment:
Will we know for sure that we'll have at least 2 values in this array?

Reply:
I guess we might not want to rely on the service's validation (if it's not a valid config string, it's discarded by the service so will be `""`) Doesn't hurt to add a length check, so I'll do that 👍